src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/IPackageReferenceProject.cs

```diff
@@ -9,6 +9,15 @@ namespace NuGet.PackageManagement.VisualStudio
 {
     public interface IPackageReferenceProject
     {
+        /// <summary>
+        /// Gets both the installed (top level) and transitive package references for this project, optionally including transitive origins for transitive packages.
+        /// Returns the package reference as two separate lists (installed and transitive).
+        /// </summary>
+        /// <param name="includeTransitiveOrigins">Set it to <c>true</c> to get transitive origins in transitive packages list</param>
+        /// <param name="token">Cancellation token</param>
+        /// <returns>A <see cref="ProjectPackages"/>A struct with two lists: installed and transitive packages</returns>
+        public Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool includeTransitiveOrigins, CancellationToken token);
```

Comment:
nit: Given that this isn't a client visible public API, we don't have to duplicate the API :) We can just have 1.

Reply:
I will address this in a follow-up issue. https://github.com/NuGet/Home/issues/12098